test/e2e/storage/drivers/in_tree.go

```diff
@@ -1303,7 +1303,8 @@ func (v *vSphereDriver) PrepareTest(ctx context.Context, f *framework.Framework)
 
 func (v *vSphereDriver) CreateVolume(ctx context.Context, config *storageframework.PerTestConfig, volType storageframework.TestVolType) storageframework.TestVolume {
 	f := config.Framework
-	nodeInfo := vspheretest.GetReadySchedulableRandomNodeInfo(ctx, f.ClientSet)
+	testContext := vspheretest.NewTestContext(f)
+	nodeInfo := vspheretest.GetReadySchedulableRandomNodeInfo(ctx, testContext, f.ClientSet)
 	volumePath, err := nodeInfo.VSphere.CreateVolume(&vspheretest.VolumeOptions{}, nodeInfo.DataCenterRef)
```

Comment:
this `CreateVolume actually connects to vSphere and files `nodeInfo.VSphere.Client`. But then both `nodeInfo` and  `testContext` get out of scope are garbage collected without closing the connection.

Reply:
I think we need to allocate some TestContext that's used in all tests and then destroyed.

Looking at the test framework, [InitVSphereDriver](https://github.com/kubernetes/kubernetes/blob/27afb7d75d20d7e986095b0d7125da60ba6322f0/test/e2e/storage/drivers/in_tree.go#L1198) is called only once [here](https://github.com/kubernetes/kubernetes/blob/27afb7d75d20d7e986095b0d7125da60ba6322f0/test/e2e/storage/in_tree_volumes.go#L80) and it's inside `Describe`. If ginkgo's `Describe` allows `DeferCleanup`, we  can both initialize the context and call `DeferCleanup` in `InitVSphereDriver`.